### DISCUSSION TITLE
It seems that ”ccache“ doesn't work with CMAKE_TOOLCHAIN_FILE “emscripten.cmake”

### DISCUSSION THREAD

**Discussion by DOOOBETTER** [emscripten-core/emscripten#23248] [Q&A] [answered]
Hello all,
when I was trying to add "ccache" to my cmake program, no new files were added to my ccache directory. But I tried test "ccache em++/emcc" in a small project, it does work.

cmake -B build_wasm -S . -DCMAKE_TOOLCHAIN_FILE=/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_BUILD_TYPE=MinSizeRel -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache  
I use this way to apply ccache to my cmake file.

**Comment by sbc100** [comment]
Then you run the build do you see `ccache` being used in the build steps?   i.e. if you do a clean build and then build with `make VERBOSE=1` do you see ccache at the start of each compile step?

It looks like ccache has a bunch of different verbosity options you can try which might help you figure out what is going on?

**Reply by DOOOBETTER** [reply]
Thank you @sbc100 , I tried several ways to fix my problem and found that adding " export CCACHE_COMPILER_TYPE=emcc" to my script file can work out

**Reply by sbc100** [reply]
Strange.. I can't find any reference to `CCACHE_COMPILER_TYPE` anywhere.  What do you think it does?  Is ccache reading it?  I don't see any mention of it in the ccache man page.

**Reply by DOOOBETTER** [reply, marked as answer]
> Strange.. I can't find any reference to `CCACHE_COMPILER_TYPE` anywhere. What do you think it does? Is ccache reading it? I don't see any mention of it in the ccache man page.

sorry, I made a mistake. The real reason why my "ccache" doesn't work is that I didn't use an absolute path to set the "CCACHE_DIR".
